pycmr/granule.py

```diff
@@ -50,6 +51,64 @@ def point(self, point=None):
 
         return self
 
+    def downloadable(self, downloadable):
+        """
+        Set the downloadable value for the query.
+
+        Must be of type Boolean
+        """
+        if not isinstance(downloadable, bool):
+            return
+
+        self.params['downloadable'] = downloadable
+
+    def online_only(self, online_only):
+        """
+        Set the online_only value for the query.
+
+        Must be of type Boolean
+        """
+
+        if not isinstance(online_only, bool):
+            return
+
+        self.params['online_only'] = online_only
+
+    def entry_title(self, entry_title):
+        """
+        Set the entry_title value for the query
+        """
+
+        entry_title = urllib.parse.quote(entry_title)
+
+        self.params['entry_title'] = entry_title
+
+    def orbit_number(self, orbit_number):
+        """"
+        Set the orbit_number value for the query
+        """
+
+        if isinstance(orbit_number, int):
```

Comment:
Although in general `isinstance()` isn't the most pythonic thing to do (kind of goes against the idea of duck typing), it makes sense to me to use it in cases where the method really can't accept anything other than a single type, like `downloadable()` or `online_only()` where it needs to be a boolean.

In this case though, it would make less sense, because we could actually support multiple types here. By using the `int()` method, we can expand the supported types to strings and floats in addition to a properly passed int. It's kind of a weird way to look at it, but it's part of the whole "if it quacks like a duck, it's a duck" approach.

So the idea here is instead of checking for the instance (which limits you to only allowing one type), use a try/catch that executes something like `int()` and catch the exception that would arise if they passed something weird through like an object or a boolean.